lib/releases.js

```diff
@@ -158,8 +158,10 @@ async function create_release(pg_pool, app_src, app_dest, build_uuid, descriptio
   release.updated = release.created = new Date();
   await create_release_record(pg_pool, release_obj_to_postgres(release))
   try {
-    await request_release(pg_pool, app_dest, build.docker_registry_url, release_id);
+    // This must come before the request release, if it doesn't the request release may
+    // hear back about a successful release before the update release status can run.
     await common.update_release_status(pg_pool, app_dest.app, release_id, 'pending');
+    await request_release(pg_pool, app_dest, build.docker_registry_url, release_id);
     setTimeout(async () => {
```

Comment:
COBRA-2501, this appears to be a race condition.  In the example I found the app released and the apps watcher notified of the successful release 1.18 seconds later, this could potentially have been before the update_release_status could execute (most likely due to the async nature and that request_release does more after the release).  In these circumstances the "pending" status would overwrite the "successful" status and the release event would fail to trigger a notification to Github.  By setting the release to pending status before the actual request it prevents the race condition from ever happening.